### ISSUE TITLE
Tensorflow Retrain Model performance

### ISSUE THREAD

**Issue by kinDSa** [tensorflow/tensorflow#8497]
I am using [Tensorflow for poet](https://codelabs.developers.google.com/codelabs/tensorflow-for-poets/#0) guide for train own model. I have create retrained_graph.pb and retrained_labels.txt. While I use it in application then I get error that 

`Caused by: java.lang.UnsupportedOperationException: Op BatchNormWithGlobalNormalization is not available in GraphDef version 21. It has been removed in version 9. Use tf.nn.batch_normalization(). at org.tensorflow.Graph.importGraphDef(Native Method) at org.tensorflow.Graph.importGraphDef(Graph.java:118) `

After That  further train model for application use [Tensorflow for mobile ](https://petewarden.com/2016/09/27/tensorflow-for-mobile-poets/)  blog and create **optimized_graph.pb, rounded_graph.pb, mmapped_graph.pb** files.

optimized_graph.pb and rounded_graph.pb file work in android application without any error.
While use mmapped_graph.pb I get error that **Failed to initialize: java.io.IOException: Not a valid TensorFlow Graph serialization: Invalid GraphDef**

Performance of application is not good while use optimized_graph.pb and rounded_graph.pb file.While application camera screen not contain any flower photos otherwise random flower name show with high confidence rate. Any way to detect only flower and remain blank when not flowers.

![screenshot](https://cloud.githubusercontent.com/assets/25680329/24044730/8edab30c-0b42-11e7-8209-a5fab382b81c.png)

**Comment by andrewharp** [comment]
Closing as this seems to be a duplicate of #8396.

Regarding the model performance: this is expected because the SoftMax layer at the top of Inception will also output values summing to 1.0. To work around this you would need to use a different type of layer for classification or add a default class to your training data (training with a variety of random images may work, but that's a discussion for elsewhere).

**Comment by ainurb** [comment]
Hi @kinDSa! I'm doing the same tutorial on Tensorflow for poets, but getting some errors on retraining model. May be it's because of tensorflow version. Tutorial was written on 0.9.0 version. And I've tried get.io/tensorflow/tensorflow:latest-devel. May I ask which version did you use? :)

**Comment by kinDSa** [comment]
@ainurb, I am using 1.0.1 version. Still I getting this error but after follow tensorflow for mobile blog then I successfully use model.
